code/game/objects/items/RPD.dm

```diff
@@ -264,6 +264,14 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 		return TRUE
 	return ..()
 
+/obj/item/pipe_dispenser/pre_attack_secondary(obj/machinery/atmospherics/target, mob/user, params)
+	if(!istype(target, /obj/machinery/atmospherics))
+		return ..()
```

Comment:
Don't return `..()`, return `SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN` in all cases, otherwise you'll end up attacking with right-click, something I eventually want to phase out.

Reply:
That was actually intentional, I didn't know we don't want attack behaviour on right click.